packages/myst-cli/src/transforms/include.ts

```diff
@@ -7,24 +7,64 @@ import type { VFile } from 'vfile';
 import { parseMyst } from '../process/myst.js';
 import type { ISession } from '../session/types.js';
 import { watch } from '../store/reducers.js';
+import { TexParser } from 'tex-to-myst';
 
-export const makeFileLoader =
-  (session: ISession, vfile: VFile, baseFile: string) => (filename: string) => {
-    const dir = path.dirname(baseFile);
-    const fullFile = path.join(dir, filename);
+/**
+ * Return resolveFile function
+ *
+ * If `sourceFile` is format .tex, `relativeFile` will be resolved relative to the
+ * original baseFile; otherwise, it will be resolved relative to `sourceFile`.
+ *
+ * The returned function will resolve the file as described above, and return it if
+ * it exists or log an error and return undefined otherwise.
+ */
+export const makeFileResolver =
+  (baseFile: string) => (relativeFile: string, sourceFile: string, vfile: VFile) => {
+    const base = sourceFile.toLowerCase().endsWith('.tex') ? baseFile : sourceFile;
+    const fullFile = path.resolve(path.dirname(base), relativeFile);
     if (!fs.existsSync(fullFile)) {
-      fileError(vfile, `Include Directive: Could not find "${fullFile}" in "${baseFile}"`, {
-        ruleId: RuleId.includeContentLoads,
-      });
+      fileError(
+        vfile,
+        `Include Directive: Could not find "${relativeFile}" relative to "${base}"`,
+        {
+          ruleId: RuleId.includeContentLoads,
+        },
+      );
       return;
     }
-    session.store.dispatch(
-      watch.actions.addLocalDependency({
-        path: baseFile,
-        dependency: fullFile,
-      }),
-    );
-    return fs.readFileSync(fullFile).toString();
+    return fullFile;
+  };
+
+/**
+ * Return loadFile function
+ *
+ * Loaded file is added to original baseFile's dependencies.
+ */
+export const makeFileLoader = (session: ISession, baseFile: string) => (fullFile: string) => {
+  session.store.dispatch(
+    watch.actions.addLocalDependency({
```

Comment:
These `include` dependencies _do not_ show up in the page json "dependency" field, but they are respected for live site reloading.

I think that's ok - the json "dependencies" are used for loading data from other pages; in the case of `include`, this data is actually moved to the page itself.

🤔 Maybe if we update `include` to allow including _partial_ snippets from md files we may need to load the rest of the data from that page...? (Similar to `embed` pulling in a single figure.) But for now, it's unnecessary.

Reply:
Agree, the included files are a local convinience, not for how the final documents work.